addon/components/async-await.js

```diff
@@ -161,15 +182,21 @@ export default Component.extend({
       rejectReason: UNINITIALIZED()
     });
 
-    Promise.resolve(promise).then(
-      bind(this, this.didResolve, promise),
-      bind(this, this.didReject, promise)
+    Promise.resolve(this.promise).then(
+      bind(this, this.didResolve, this.promise),
+      bind(this, this.didReject, this.promise)
     );
   },
 
   didResolve(resolvedPromise, value) {
     if (this.shouldIgnorePromise(resolvedPromise)) { return; }
 
+    if (this.promiseKeys) {
+      let hash = {};
+      this.promiseKeys.forEach((k, i) => hash[k] = value[i]);
+      value = hash;
+    }
+
```

Comment:
This diff looks bigger than necessary. Once we normalize it upfront, it shouldn't affect the rest of the code.

```js
{
  // ...

  didReceivePromise(promise) {
    if (promise === this.awaited) { return; }

    // added
    if (isHash(promise)) {
      promise = RSVP.hash(promise);
    }
    // end added

    // the rest shouldn't have to change, other than any refactor/renaming you might want to do
    // ...
  },

  // ...
```